### PR TITLE
Added startsWith/endsWith implementations for character prefix/suffix…

### LINKED ISSUE #4252: RFC: should startsWith and endsWith work with characters?
From a consistency perspective, it seems logical to me that you can do this:

```nim
doAssert "abc".startsWith("a") # works
doAssert "abc".endsWith("c") # works
doAssert "abc".startsWith('a') # compile error
doAssert "abc".endsWith('c') # compile error
```

The latter can easily be achieved with:

```nim
doAssert "abc"[0] == 'a'
var s = "abc"
doAssert s[s.high] == 'c'
```

however, this feels a bit more messy to me and doesn't as obviously show the intention.

### PR DESCRIPTION
**Pull request by abudden**
… (fixes #4252).

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -862,6 +862,10 @@ proc startsWith*(s, prefix: string): bool {.noSideEffect,
     if s[i] != prefix[i]: return false
     inc(i)
 
+proc startsWith*(s: string, prefix: char): bool {.noSideEffect.} =
```
Comment (Araq) [inline lib/pure/strutils.nim]: Mark as 'inline'.

File: lib/pure/strutils.nim [hunk 2]
```diff
@@ -874,6 +878,10 @@ proc endsWith*(s, suffix: string): bool {.noSideEffect,
     inc(i)
   if suffix[i] == '\0': return true
 
+proc endsWith*(s: string, suffix: char): bool {.noSideEffect.} =
```
Comment (Araq) [inline lib/pure/strutils.nim]: Mark as 'inline'.

File: lib/pure/strutils.nim (hunk 1 above)
Comment (abudden) [inline lib/pure/strutils.nim]: Done.

File: lib/pure/strutils.nim (hunk 2 above)
Comment (abudden) [inline lib/pure/strutils.nim]: Done.

**Pull request merged**